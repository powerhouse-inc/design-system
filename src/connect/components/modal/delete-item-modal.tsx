import { Modal, ModalProps } from '@/powerhouse/components/modal';
import React from 'react';
import { Button } from 'react-aria-components';
import { twMerge } from 'tailwind-merge';

const buttonStyles =
    'min-h-[48px] min-w-[142px] text-base font-semibold py-3 px-6 rounded-xl outline-none active:opacity-75 hover:scale-105 transform transition-all';

export interface ConnectDeleteItemModalProps extends ModalProps {
    header: React.ReactNode;
    body: React.ReactNode;
    onDelete: () => void;
    cancelLabel: string;
    deleteLabel: string;
}

export const ConnectDeleteItemModal = (props: ConnectDeleteItemModalProps) => {
    const {
        body,
        header,
        onClose,
        onDelete,
        cancelLabel,
        deleteLabel,
        modalProps = {},
        dialogProps = {},
        ...restProps
    } = props;

    const { className: modalClassName, ...restModalProps } = modalProps;
    const { className: dialogClassName, ...restDialogProps } = dialogProps;

    return (
        <Modal
            modalProps={{
                className: twMerge(
                    'top-10',
                    typeof modalClassName === 'string' && modalClassName,
                ),
                ...restModalProps,
            }}
            dialogProps={{
                className: twMerge('rounded-3xl', dialogClassName),
                ...restDialogProps,
            }}
            onClose={onClose}
            {...restProps}
        >
            <div className="w-[465px] py-12 px-8 text-[#565868]">
                <div className="text-2xl font-semibold text-center">
                    {header}
                </div>
                <div className="text-xl mt-8 text-center">{body}</div>
                <div className="mt-8 flex justify-center gap-3">
                    <Button
                        onPress={onClose}
                        className={twMerge(
                            buttonStyles,
                            'bg-[#E8ECEFBF] text-[#141718]',
                        )}
                    >
                        {cancelLabel}
                    </Button>
                    <Button
                        onPress={onDelete}
                        className={twMerge(
                            buttonStyles,
                            'bg-[#EA4335] text-[#FEFEFE]',
                        )}
                    >
                        {deleteLabel}
                    </Button>
                </div>
            </div>
        </Modal>
    );
};
