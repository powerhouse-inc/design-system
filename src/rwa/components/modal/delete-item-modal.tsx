import { Icon, Modal } from '@/powerhouse';
import React, { ComponentPropsWithoutRef, Fragment } from 'react';

export type RWADeleteItemModalProps = ComponentPropsWithoutRef<typeof Modal> & {
    itemName: React.ReactNode;
    dependentItemName: React.ReactNode;
    dependentItemList: React.ReactNode[];
    onContinue: () => void;
};

export const RWADeleteItemModal = (props: RWADeleteItemModalProps) => {
    const {
        itemName,
        dependentItemName,
        dependentItemList,
        onOpenChange,
        onContinue,
        ...restProps
    } = props;

    return (
        <Modal
            overlayProps={{
                className: 'top-10',
            }}
            contentProps={{
                className: 'rounded-3xl',
            }}
            onOpenChange={onOpenChange}
            {...restProps}
        >
            <div className="w-[400px] p-6 text-slate-300">
                <div className="border-b border-slate-50 pb-2 text-2xl font-bold text-gray-800">
                    Cannot delete {itemName}
                </div>
                <div className="my-6 flex gap-2 rounded-md bg-orange-100 p-4 text-orange-800">
                    <div>
                        <Icon name="error" className="mt-1 text-orange-800" />
                    </div>
                    <div>
                        Warning! Cannot delete this {itemName} because there are{' '}
                        {dependentItemName} that depend on it. Please change or
                        delete those first.
                    </div>
                </div>
                <div className="my-6 rounded-md bg-slate-50 p-4 text-slate-200">
                    {dependentItemList.map((item, index) => (
                        <Fragment key={index}>{item}</Fragment>
                    ))}
                </div>
                <div className="mt-8 flex justify-between gap-3">
                    <button
                        onClick={onContinue}
                        className="min-h-12 min-w-36 flex-1 rounded-xl bg-gray-800 px-6 py-3 text-base font-semibold text-gray-50 outline-none transition-all hover:scale-105 active:opacity-75"
                    >
                        Back
                    </button>
                </div>
            </div>
        </Modal>
    );
};
