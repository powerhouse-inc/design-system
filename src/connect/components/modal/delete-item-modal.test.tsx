import { fireEvent, render, screen } from '@testing-library/react';
import { ConnectDeleteItemModal } from './delete-item-modal';

describe('Modal Component', () => {
    it('should match snapshot', () => {
        render(
            <ConnectDeleteItemModal
                open={true}
                onClose={() => {}}
                data-testid="delete-modal"
                body="Are you sure you want to delete this folder? All files and subfolders within it will be removed."
                cancelLabel="Cancel"
                deleteLabel="Delete"
                header="Delete “Ecosystem Actors” folder?"
                onDelete={() => {}}
            />,
        );

        const modalComponent = screen.getByTestId('delete-modal');

        expect(modalComponent).toMatchSnapshot();
    });

    it('should render modal content', () => {
        const headerText = 'Delete “Ecosystem Actors” folder?';
        const bodyText =
            'Are you sure you want to delete this folder? All files and subfolders within it will be removed.';
        const cancelLabel = 'Cancel';
        const deleteLabel = 'Delete';

        render(
            <ConnectDeleteItemModal
                open={true}
                onClose={() => {}}
                data-testid="delete-modal"
                body={bodyText}
                cancelLabel={cancelLabel}
                deleteLabel={deleteLabel}
                header={headerText}
                onDelete={() => {}}
            />,
        );

        const header = screen.getByText(headerText);
        const body = screen.getByText(bodyText);
        const cancelButton = screen.getByText(cancelLabel);
        const deleteButton = screen.getByText(deleteLabel);

        expect(header).toBeInTheDocument();
        expect(body).toBeInTheDocument();
        expect(cancelButton).toBeInTheDocument();
        expect(deleteButton).toBeInTheDocument();
    });

    it('should trigger onClose callback when cancel is clicked', () => {
        const onClose = jest.fn();
        const cancelLabel = 'Cancel';

        render(
            <ConnectDeleteItemModal
                open={true}
                onClose={onClose}
                data-testid="delete-modal"
                body="Are you sure you want to delete this folder? All files and subfolders within it will be removed."
                cancelLabel={cancelLabel}
                deleteLabel="Delete"
                header="Delete “Ecosystem Actors” folder?"
                onDelete={() => {}}
            />,
        );

        fireEvent.click(screen.getByText(cancelLabel));

        expect(onClose).toHaveBeenCalled();
    });

    it('should trigger onDelete callback when CTA is clicked', () => {
        const onDelete = jest.fn();
        const deleteLabel = 'Delete';

        render(
            <ConnectDeleteItemModal
                open={true}
                onClose={() => {}}
                data-testid="delete-modal"
                body="Are you sure you want to delete this folder? All files and subfolders within it will be removed."
                cancelLabel="Cancel"
                deleteLabel={deleteLabel}
                header="Delete “Ecosystem Actors” folder?"
                onDelete={onDelete}
            />,
        );

        fireEvent.click(screen.getByText(deleteLabel));

        expect(onDelete).toHaveBeenCalled();
    });
});
