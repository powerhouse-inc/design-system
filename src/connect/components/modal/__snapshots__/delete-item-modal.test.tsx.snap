// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should match snapshot 1`] = `
<div
  class="bg-overlay fixed inset-0 flex justify-center items-start animate-in fade-in duration-300 ease-out"
  data-entering="true"
  data-rac=""
  data-testid="delete-modal"
  style="--visual-viewport-height: 768px;"
>
  <div
    class="flex justify-center relative top-10 animate-in zoom-in-95 ease-out duration-300"
    data-entering="true"
    data-rac=""
  >
    <div
      style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
    >
      <button
        aria-label="Dismiss"
        id="react-aria-:r0:"
        tabindex="-1"
      />
    </div>
    <section
      class="bg-white outline-none rounded-3xl"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="w-[465px] py-12 px-8 text-[#565868]"
      >
        <div
          class="text-2xl font-semibold text-center"
        >
          Delete “Ecosystem Actors” folder?
        </div>
        <div
          class="text-xl mt-8 text-center"
        >
          Are you sure you want to delete this folder? All files and subfolders within it will be removed.
        </div>
        <div
          class="mt-8 flex justify-center gap-3"
        >
          <button
            class="min-h-[48px] min-w-[142px] text-base font-semibold py-3 px-6 rounded-xl outline-none active:opacity-75 hover:scale-105 transform transition-all bg-[#E8ECEFBF] text-[#141718]"
            data-rac=""
            type="button"
          >
            Cancel
          </button>
          <button
            class="min-h-[48px] min-w-[142px] text-base font-semibold py-3 px-6 rounded-xl outline-none active:opacity-75 hover:scale-105 transform transition-all bg-[#EA4335] text-[#FEFEFE]"
            data-rac=""
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
    </section>
  </div>
</div>
`;
